WordPress/gutenberg
# Navigation: refactor to TypeScript

## What?


Refactor the `Navigation` component to TypeScript

## Why?


See #35744

## How?


See [the contributing guidelines](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript)

Reviewing commit-by-commit may be easier.

## Testing Instructions





- Check that the few runtime code changes make sense
- Tests pass
- Storybook component works as expected
- ~Editor implementation of `Navigation` works as expected~ there doesn't seem to be any leftover usage of the Navigation component
